src/components/SaveExport/ExportDialog.jsx

```diff
@@ -104,32 +111,29 @@ export default () => {
         }
     }, [isExportDialogVisible]);
 
-    const end = windowEnd || options.timestamp;
-    const begin = windowBegin || end - windowDuration;
-
-    const [from, to] =
-        cursorBegin === null ? [begin, end] : [cursorBegin, cursorEnd];
-
-    const indexBegin = Math.ceil(timestampToIndex(from));
-    const indexEnd = Math.floor(timestampToIndex(to));
-    const numberOfRecords = indexEnd - indexBegin + 1;
-    const filesize = calculateTotalSize(contentSelection, numberOfRecords);
+    useEffect(() => {
+        const records = indexEnd - indexBegin + 1;
```

Comment:
Seems like `indexBegin` and `indexEnd` can be null, so we should probably add a guard clause for this? E.g. 

```
if (!indexBegin || !indexEnd) return;
```

to prevent potential `NaN` issues.

Reply:
Seems to be a breaking change. Could we instead do:
`const numberOfRecords = indexEnd - indexBegin + 1 || 0;` ?